src/streamlink/plugins/niconicochannelplus.py

```diff
@@ -0,0 +1,67 @@
+"""
+$description NicoNico Channel Plus (nicochannel+) is a new feature of Niconico Channel.
+$url nicochannel.jp
+$type vod
+$account Not required. Full-length videos are accessible.
+$notes Downloading non-free content is not recommended.
+"""
+
+import json
+import logging
+import re
+
+from streamlink.plugin import Plugin, pluginmatcher
+from streamlink.plugin.api import HTTPSession
+from streamlink.stream.hls import HLSStream
+
+log = logging.getLogger(__name__)
+
+
+@pluginmatcher(
+    re.compile(
+        r'^https?://nicochannel\.jp/(?P<channel>[a-z0-9_-]+)/video/(?P<id>sm[a-zA-Z0-9]+)$'
+    )
+)
+class NicoNicoChannelPlus(Plugin):
+    def get_video_page_info(self, http: HTTPSession, video_id: str) -> dict:
+        video_page_json = json.loads(
+            http.get(
+                url=f'https://nfc-api.nicochannel.jp/fc/video_pages/{video_id}',
+            ).content
+        )['data']['video_page']
+
+        return video_page_json
```

Comment:
Streamlink uses validation schemas for extracting data from JSON or any other response format, to prevent decoding issues and for being able to have better error messages when something breaks or something else/unexpected gets returned.

Calling `json.loads()` and accessing data via `__getitem__` will lead to random `LookupError`s being raised (`KeyError` or `IndexError`). Reading `Response.content` is also not the right coice because it returns `bytes` instead of a unicode string via `Response.text` that gets decoding accordingly from the HTTP headers or the guessing of chardet/charset-normalizer, so `json.loads()` would have to decode it on its own, which is bad.

```suggestion
        return self.session.http.get(
            self._URL_API_VIDEO_PAGES.format(video_id=video_id),
            schema=validate.Schema(
                validate.parse_json(),
                {"data": {"video_page": {"title": str}}},
                validate.get(("data", "video_page")),
            ),
        )
```

Reply:
Thanks, I didn't know "validation schemas" before. It has been added in 4d132c87f7f9c0f307e1817be4f78b9b11aec758 and is more safe now.